src/js/components/modals/UpdateConfigModal.js

```diff
@@ -0,0 +1,249 @@
+import classNames from 'classnames';
+import {Modal} from 'reactjs-components';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import AdvancedConfig from '../AdvancedConfig';
+import CosmosErrorMessage from '../CosmosErrorMessage';
+import CosmosPackagesStore from '../../stores/CosmosPackagesStore';
+import InternalStorageMixin from '../../mixins/InternalStorageMixin';
+import Service from '../../structs/Service';
+import StringUtil from '../../utils/StringUtil';
+import UniversePackage from '../../structs/UniversePackage';
+import Util from '../../utils/Util';
+
+const METHODS_TO_BIND = [
+  'handleAdvancedFormChange',
+  'handleConfigSave',
+  'handleErrorConfigEditClick',
+  'handleModalClose',
+  'getConfigSubmit'
+];
+
+class UpdateConfigModal extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.internalStorage_set({
+      hasFormErrors: false,
+      updateError: null,
+      pendingRequest: false,
+      updateSuccess: false
+    });
+
+    this.store_listeners = [{
+      name: 'cosmosPackages',
+      events: [
+        'installError',
+        'installSuccess'
+      ]
+    }];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onCosmosPackagesStoreInstallSuccess() {
+    this.internalStorage_update({
+      updateError: null,
```

Comment:
i cannot for the life of me see where `updateError` ever gets set to anything other than `null`